packages/hash/api/src/graphql/resolvers/knowledge/page/set-parent-page.ts

```diff
@@ -0,0 +1,46 @@
+import { ApolloError } from "apollo-server-express";
+
+import { LoggedInGraphQLContext } from "../../../context";
+import {
+  MutationSetParentKnowledgePageArgs,
+  ResolverFn,
+} from "../../../apiTypes.gen";
+import {
+  mapPageModelToGQL,
+  UnresolvedKnowledgePageGQL,
+} from "../model-mapping";
+import { PageModel } from "../../../../model";
+
+export const setParentKnowledgePage: ResolverFn<
+  Promise<UnresolvedKnowledgePageGQL>,
+  {},
+  LoggedInGraphQLContext,
+  MutationSetParentKnowledgePageArgs
+> = async (
+  _,
+  { pageEntityId, parentPageEntityId, prevIndex = null, nextIndex = null },
+  { dataSources: { graphApi }, user },
+) => {
+  if (pageEntityId === parentPageEntityId) {
+    throw new ApolloError("A page cannot be the parent of itself");
+  }
+
+  const pageModel = await PageModel.getPageById(graphApi, {
+    entityId: pageEntityId,
+  });
+
+  const newParentPage = parentPageEntityId
```

Comment:
For consistency these should all be `newParentPageModel` and such, right?

Reply:
Addressed in https://github.com/hashintel/hash/pull/1157/commits/dc6134dfe601d43f8b534b0b38019276adf29aea